test/e2e-tests.sh

```diff
@@ -121,6 +121,23 @@ go_test_e2e -timeout=2m ./test/e2e/initcontainers ${TEST_OPTIONS} || failed=1
 toggle_feature kubernetes.podspec-init-containers Disabled
 toggle_feature kubernetes.podspec-volumes-emptydir Disabled
 
+# RUN PVC tests with default storage class if available.
+if (( USE_DEFAULT_STORAGE )) ; then
+  kubectl apply -f "${E2E_YAML_DIR}/test/config/pvc/pvc.yaml" -n serving-tests
```

Comment:
Rather than applying this here should we just include this file when installing knative and everything else? 

This would mean adding an `if` to `e2e-common.sh` `install` function

Reply:
> I still think you need the PVC applied otherwise the pod references a resource that doesn't exist. So I don't really see the need for USE_DEFAULT_STORAGE
Reason is to skip the pvc we ship and allow to use some other pvc provided by whoever is running the tests. For example I might have a K8s cluster where I might want to use some other storage class in the pvc definition.

Reply:
We could but I want to apply the resource only at the place it is needed.

Reply:
One reason why I use kapp to manage all the resources is because it has wait logic and will fail if the resource is not ready by some duration.

So I would rather stick with that pattern instead of re-introducing `kubectl` around these scripts.

The other option would be to have the test create the PVC and wait for it to be ready

Reply:
Ok I will take a look at the second option to see how that would work.